Open Judge System/OJS.Services/Business/Problems/ProblemsBusinessService.cs

```diff
@@ -38,5 +63,52 @@ public void RetestById(int id)
                 scope.Complete();
             }
         }
+
+        public void DeleteById(int id)
+        {
+            var problem = this.problemsData
+                .GetByIdQuery(id)
+                .Select(p => new
+                {
+                    p.ProblemGroupId,
+                    p.ProblemGroup.ContestId
+                })
+                .FirstOrDefault();
+
+            if (problem == null)
+            {
+                return;
+            }
+
+            this.testRunsData.DeleteByProblem(id);
+
+            this.problemResourcesData.DeleteByProblem(id);
+
+            this.submissionsData.DeleteByProblem(id);
+
+            this.problems.Delete(id);
+            this.problems.SaveChanges();
+
+            if (!this.contestsData.IsOnlineById(problem.ContestId))
```

Comment:
Would this work after the problem is deleted? does problemsData.IsFromOnlineContestById work with deleted entities?

Reply:
I fixed it, now this is `contestsData`